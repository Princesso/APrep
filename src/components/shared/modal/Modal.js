import React from 'react'
import './modal.css'

const Modal = ({
  toggleModalView,
  children
}) => {
  return(
    <div className="modal-bg">
      <div onClick={toggleModalView} className="overlay" />
      <div className="modal">
        {children}
        <button onClick={() => toggleModalView()} className="close-modal">close</button>
      </div>
    </div>
  )
}

export default Modal