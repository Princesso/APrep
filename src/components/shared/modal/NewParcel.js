import React from 'react'
import Modal from './Modal'
import {createParcel} from '../../../actions/userActions'
import {connect} from 'react-redux'

class NewParcel extends React.Component {
  state = {
    itemName: '',
    weight: 0,
    fromAddress: '',
    toAddress: '',
    phoneNumber: '',
    recipientName: ''
  }

  handleChange = e => this.setState({ [e.target.name]: e.target.value });

  handleSubmit= async(e) => {
    e.preventDefault();
    const newDeliveryData = this.state;
    //write validation function
    this.setState({
      itemName: '',
      weight: 0,
      fromAddress: '',
      toAddress: '',
      phoneNumber: '',
      recipientName: ''
    })
    this.props.createParcel(newDeliveryData)
  }

  render(){
    const { toggleModalView, visible } = this.props;
    const { itemName, weight, fromAddress, toAddress, phoneNumber, recipientName } = this.state
    return (
      <div>
        {visible && 
        <Modal toggleModalView={toggleModalView} className="modal">       
          <div className="form-holder">
            <form className="form" method="post" onSubmit={this.handleSubmit}> 
              <h3>New Delivery Order</h3>
              <input 
                type="text" 
                placeholder="Item Name" 
                id="item-name"
                name="itemName"
                value={itemName}
                onChange={this.handleChange}
              />
              <input 
                type="number" 
                placeholder="Approximate Weight in Kilograms" 
                id="weight"
                name="weight"
                value={weight}
                onChange={this.handleChange}
              />
              <input 
                type="text" 
                placeholder="Parcel Pick-up Location" 
                autoComplete="address-level2" 
                id="from-address"
                name="fromAddress"
                value={fromAddress}
                onChange={this.handleChange}
              />
              <input 
                type="text" 
                placeholder="Recipient's Address" 
                autoComplete="address-level2" 
                id="to-address"
                name="toAddress"
                value={toAddress}
                onChange={this.handleChange}
              />
              <input 
                type="number" 
                placeholder="Recipient's phone number" 
                id="recipient-phone"
                name="phoneNumber"
                value={phoneNumber}
                onChange={this.handleChange}
                />
              <input 
                type="text" 
                placeholder="Recipient's Name" 
                id="recipient-name"
                name="recipientName"
                value={recipientName}
                onChange={this.handleChange}
              />
              <p id="created"></p>
              <div className="send-parcel-button-div"><button type="submit" className="send-parcel" id="send-parcel-button">Send Parcel</button></div>
            </form>
          </div>
        </Modal>
        }
      </div>
    )
  }
}
const mapStateToProps = state => ({
  newParcel: state.newParcel
})

export default connect(mapStateToProps, {createParcel})(NewParcel)