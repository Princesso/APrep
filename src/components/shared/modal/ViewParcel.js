import React from 'react'
import Modal from './Modal'
import {Link} from 'react-router-dom'

class ViewParcel extends React.Component{
  render(){
    const { visible, toggleModalView, selectedUserParcel } = this.props;
    return (
      <div>
        {visible && 
        <Modal toggleModalView={toggleModalView} className="modal">      
          <div className="form-holder">
          <Link to="#" className="linker">Track Order</Link><br/> 
            <form className="form"> 
              <p><span className="title">Order ID : </span><span className="info" >{selectedUserParcel.id}</span></p><br/><hr/>
              <p><span className="title">Placed At : </span><span className="info" >{selectedUserParcel.fromaddress}</span></p>
              <p><span className="title">Current Location : </span><span className="info" >{selectedUserParcel.currentlocation}</span></p><br/> 
              <p><span className="title">Destination : </span><span className="info" ></span>{selectedUserParcel.toaddress}</p><br/> <hr/>
              <p><span className="title">Item Weight : </span><span className="info" ></span>{selectedUserParcel.weight} kg</p><br/> 
              <p><span className="title">Order Status :</span><span className="info" ></span>{selectedUserParcel.status}</p><br/>  <hr/>
              <p><span className="title">Approximate Price : </span><span className="info"></span>{selectedUserParcel.id}<br/></p>
              <div className="view-parcel-buttons-div">
              <Link to="/"> <button className="change-destination">Edit</button> </Link>
              <div></div>
              <Link to="/"><button className="cancel-order">Cancel Order</button></Link>
           </div>
          </form>
          </div>
        </Modal>
        }
      </div>
    )
  }
}

export default ViewParcel