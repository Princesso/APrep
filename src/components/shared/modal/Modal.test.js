import React from 'react';
import Modal from './Modal';
import { NewParcel } from './NewParcel';
import ViewParcel from './ViewParcel';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import Store from '../../../store';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow } from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

test('Modal', () => {
  const wrapper = shallow(
    <Provider store={Store}>
      <BrowserRouter>
          <Switch>
            <Modal> 
              <NewParcel/>
              <ViewParcel/>
            </Modal>
          </Switch>
      </BrowserRouter>
    </Provider>
  )
  expect(wrapper).toMatchSnapshot();
})


test('it should render new parcel component', ()=> {
  const wrapper = mount(
    <NewParcel/>
  )
  expect(wrapper).toMatchSnapshot();
});

test('it should render View parcel component', ()=> {
  const tree = renderer.create(
    <Provider store={Store}>
      <BrowserRouter>
        <ViewParcel/>
      </BrowserRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('it should call toggle modal when send button is clicked', ()=> {
  const toggleModalView = jest.fn()
  const props = {
    toggleModalView,
    children: ViewParcel
  }
  const wrapper = mount(<Modal { ...props}/>)
  wrapper.find('button').simulate('click');
  expect(toggleModalView).toHaveBeenCalled();
});
