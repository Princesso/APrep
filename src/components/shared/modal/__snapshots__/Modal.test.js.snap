// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal 1`] = `ShallowWrapper {}`;

exports[`it should render View parcel component 1`] = `<div />`;

exports[`it should render new parcel component 1`] = `ReactWrapper {}`;
